ios/Private/SentryOptions+RNOptions.mm

```diff
@@ -0,0 +1,210 @@
+        self.integrations = [options[@"integrations"] filteredArrayUsingPredicate:isNSString];
+    }
+
+    if ([options[@"sampleRate"] isKindOfClass:[NSNumber class]]) {
+        self.sampleRate = options[@"sampleRate"];
+    }
+
+    [self setBool:options[@"enableAutoSessionTracking"]
+            block:^(BOOL value) { _self.enableAutoSessionTracking = value; }];
+
+    [self setBool:options[@"enableOutOfMemoryTracking"]
+            block:^(BOOL value) { _self.enableOutOfMemoryTracking = value; }];
+
+    if ([options[@"sessionTrackingIntervalMillis"] isKindOfClass:[NSNumber class]]) {
+        self.sessionTrackingIntervalMillis =
+            [options[@"sessionTrackingIntervalMillis"] unsignedIntValue];
+    }
+
+    [self setBool:options[@"attachStacktrace"]
+            block:^(BOOL value) { _self.attachStacktrace = value; }];
+
+    [self setBool:options[@"stitchAsyncCode"]
+            block:^(BOOL value) { _self.stitchAsyncCode = value; }];
+
+    if ([options[@"maxAttachmentSize"] isKindOfClass:[NSNumber class]]) {
+        self.maxAttachmentSize = [options[@"maxAttachmentSize"] unsignedIntValue];
+    }
+
+    [self setBool:options[@"sendDefaultPii"]
+            block:^(BOOL value) { _self.sendDefaultPii = value; }];
+
+    [self setBool:options[@"enableAutoPerformanceTracking"]
+            block:^(BOOL value) { _self.enableAutoPerformanceTracing = value; }];
+
+    [self setBool:options[@"enableCaptureFailedRequests"]
+            block:^(BOOL value) { _self.enableCaptureFailedRequests = value; }];
+
+#if SENTRY_HAS_UIKIT
+    [self setBool:options[@"enableUIViewControllerTracking"]
+            block:^(BOOL value) { _self.enableUIViewControllerTracing = value; }];
+
+    [self setBool:options[@"attachScreenshot"]
+            block:^(BOOL value) { _self.attachScreenshot = value; }];
+
+    [self setBool:options[@"attachViewHierarchy"]
+            block:^(BOOL value) { _self.attachViewHierarchy = value; }];
+
+    [self setBool:options[@"enableUserInteractionTracing"]
+            block:^(BOOL value) { _self.enableUserInteractionTracing = value; }];
+
+    if ([options[@"idleTimeout"] isKindOfClass:[NSNumber class]]) {
+        self.idleTimeout = [options[@"idleTimeout"] doubleValue];
+    }
+
+    [self setBool:options[@"enablePreWarmedAppStartTracking"]
+            block:^(BOOL value) { _self.enablePreWarmedAppStartTracing = value; }];
+#endif
+
+    [self setBool:options[@"enableAppHangTracking"]
+            block:^(BOOL value) { _self.enableAppHangTracking = value; }];
+
+    if ([options[@"appHangTimeoutInterval"] isKindOfClass:[NSNumber class]]) {
+        self.appHangTimeoutInterval = [options[@"appHangTimeoutInterval"] doubleValue];
+    }
+
+    [self setBool:options[@"enableNetworkTracking"]
+            block:^(BOOL value) { _self.enableNetworkTracking = value; }];
+
+    [self setBool:options[@"enableFileIOTracing"]
+            block:^(BOOL value) { _self.enableFileIOTracing = value; }];
```

Comment:
M: Does this exist on RN already?

Reply:
No, this was just copied from the `cocoa` constructor. I will remove the whole `RNOption` now when the `initWIthDict` is back.